src/Microsoft.AspNetCore.Mvc.DataAnnotations/Internal/DataAnnotationsMetadataProvider.cs

```diff
@@ -80,23 +95,18 @@ public void CreateDisplayMetadata(DisplayMetadataProviderContext context)
                 displayMetadata.DataTypeName = DataType.Html.ToString();
             }
 
-            // Description
-            if (displayAttribute != null)
-            {
-                displayMetadata.Description = () => displayAttribute.GetDescription();
-                displayMetadata.Placeholder = () => displayAttribute.GetPrompt();
-            }
-
             // DisplayFormatString
             if (displayFormatAttribute != null)
             {
                 displayMetadata.DisplayFormatString = displayFormatAttribute.DataFormatString;
             }
 
-            // DisplayName
+            // Display
             if (displayAttribute != null)
             {
-                displayMetadata.DisplayName = () => displayAttribute.GetName();
+                displayMetadata.Description = LocalizeAttributeProperty(displayAttribute.GetDescription, localizer);
+                displayMetadata.Placeholder = LocalizeAttributeProperty(displayAttribute.GetPrompt, localizer);
+                displayMetadata.DisplayName = LocalizeAttributeProperty(displayAttribute.GetName, localizer);
```

Comment:
Unconditionally using `displayAttribute.GetName()` is probably _not_ what we want here. For one thing, it's inconsistent with how we localize `ValidationAttribute`s. You basically want to mimic the condition in [ValidationAttributeAdapter<TAttribute>](https://github.com/aspnet/Mvc/blob/dev/src/Microsoft.AspNetCore.Mvc.DataAnnotations/Internal/ValidationAttributeAdapterOfTAttribute.cs#L74-L77).

Put another way, there are four cases
1. Attribute is constructed with a `null` error message. Leave it alone.
2. Attribute is localized using regular `ResourceType` approach. Use that localized string unchanged. Absolutely call `GetName()` and so on in this case.
3. Attribute isn't old-fashioned localized using `ResourceType` but `localizer` is `null`. Use attribute's string (perhaps localized if it's the default string) unchanged.
4. Attribute isn't old-fashioned localized using `ResourceType` and `localizer` is non-`null`. Use the `IStringLocalizerFactory` for all it's worth. I believe `Name` and `GetName()` always return the same values when `ResourceType == null` but please double-check.

Note `ResourceType` affects all of `Description`, `Name`, and `Prompt`.